### PR TITLE
Move printing to a pass

### PR DESCRIPTION
**Pull request by kripken**
As suggested in #172.

This makes things more modular, and easier to add custom printing options (printing with name stripped, print sexpr vs. the final text format, etc.).

### CODE REVIEW

**Comment by kripken** [comment]
Should be good to go, comments welcome.

File: src/binaryen-shell.cpp [hunk 1]
```diff
@@ -281,7 +282,7 @@ static void run_asserts(size_t* i, bool* checked, AllocatingModule* wasm,
         Colors::bold(std::cout);
         std::cerr << "printing in module invalidity test:\n";
         Colors::normal(std::cout);
```
Comment (jfbastien) [inline src/binaryen-shell.cpp]: Color handling to cout seems wrong? Not related to this patch though.

File: src/passes/Print.cpp [hunk 2]
```diff
@@ -0,0 +1,490 @@
+/*
+ * Copyright 2015 WebAssembly Community Group participants
```
Comment (jfbastien) [inline src/passes/Print.cpp]: 2016

File: test/example/find_div0s.cpp [hunk 3]
```diff
@@ -45,7 +46,8 @@ int main() {
         // Check if the right operand is a constant, and if it is 0
         auto right = curr->right->dyn_cast<Const>();
         if (right && right->value.getInteger() == 0) {
-          std::cout << "We found that " << curr->left << " is divided by zero\n";
+          std::cout << "We found that ";
+          printWasm(curr->left, std::cout) << " is divided by zero\n";
```
Comment (jfbastien) [inline test/example/find_div0s.cpp]: Can't you continue chaining this since printWasm returns a stream ref?

**Comment by jfbastien** [comment]
Nice :-)

Minor comments, looks good otherwise.

File: src/binaryen-shell.cpp (hunk 1 above)
Comment (kripken) [inline src/binaryen-shell.cpp]: Hmm yeah, that's wrong. Fixing.

File: src/passes/Print.cpp (hunk 2 above)
Comment (kripken) [inline src/passes/Print.cpp]: darn time and it's slow inexorable march

File: test/example/find_div0s.cpp (hunk 3 above)
Comment (kripken) [inline test/example/find_div0s.cpp]: Almost, the order would be wrong. What is really needed is to overload the `<<` operator I suppose, in the `std` namespace? I'm always wary about doing that...

Comment (kripken) [inline test/example/find_div0s.cpp]: (overload it on some new object, of course, not a call)

**Comment by kripken** [comment]
To fix the unrelated cout/cerr issue, I realized that we don't even need the `--print-after|before` flags now that we have `--print`, which is better since you can put it in any order you want. Bonus commit removes them.

**Comment by jfbastien** [comment]
Cool, lgtm

**Comment by kripken** [comment]
Also added a commit which overloads in `std`. This lets us do that one line thing.

**Pull request merged**